File: modules/angular2/src/tools/tools.dart
library angular2.src.tools.tools;

import 'dart:js';
import 'package:angular2/angular2.dart' show ApplicationRef;
import 'common_tools.dart' show AngularTools;

/**
 * Enabled Angular 2 debug tools that are accessible via your browser's
 * developer console.
 *
 * Usage:
 *
 * 1. Open developer console (e.g. in Chrome Ctrl + Shift + j)
 * 1. Type `ng.` (usually the console will show auto-complete suggestion)
 * 1. Try the change detection profiler `ng.profiler.timeChangeDetection()`
 *    then hit Enter.
 */
void enableDebugTools(ApplicationRef appRef) {
  final tools = new AngularTools(appRef);
  context['ng'] = new JsObject.jsify({
    'profiler': {
      'timeChangeDetection': ([config]) {
        tools.profiler.timeChangeDetection(config);
      }
    }
  });
}

/**
 * Disables Angular 2 tools.
 */
void disableDebugTools() {
  context.deleteProperty('ng');
}

File: modules/angular2/src/tools/tools.ts
import {global} from 'angular2/src/core/facade/lang';
import {ApplicationRef} from 'angular2/angular2';
import {AngularTools} from './common_tools';

var context = <any>global;

/**
 * Enabled Angular 2 debug tools that are accessible via your browser's
 * developer console.
 *
 * Usage:
 *
 * 1. Open developer console (e.g. in Chrome Ctrl + Shift + j)
 * 1. Type `ng.` (usually the console will show auto-complete suggestion)
 * 1. Try the change detection profiler `ng.profiler.timeChangeDetection()`
 *    then hit Enter.
 */
export function enableDebugTools(appRef: ApplicationRef): void {
  context.ng = new AngularTools(appRef);
}

/**
 * Disables Angular 2 tools.
 */
export function disableDebugTools(): void {
  context.ng = undefined;
}
